python/ql/lib/semmle/python/dataflow/new/internal/FlowSummaryImplSpecific.qll

```diff
@@ -128,13 +128,33 @@ SummaryComponent interpretComponentSpecific(AccessPathToken c) {
   )
 }
 
-/** Gets the textual representation of a summary component in the format used for flow summaries. */
-string getComponentSpecific(SummaryComponent sc) {
-  sc = TContentSummaryComponent(any(ListElementContent c)) and
-  result = "ListElement"
+private string getContentSpecific(Content cs) {
+  cs = TListElementContent() and result = "ListElement"
+  or
+  cs = TSetElementContent() and result = "SetElement"
+  or
+  exists(int index |
+    cs = TTupleElementContent(index) and result = "TupleElement[" + index.toString() + "]"
+  )
+  or
+  exists(string key |
+    cs = TDictionaryElementContent(key) and result = "DictionaryElement[" + key + "]"
+  )
+  or
+  cs = TDictionaryElementAnyContent() and result = "DictionaryElementAny"
+  or
+  exists(string attr | cs = TAttributeContent(attr) and result = "Attribute[" + attr + "]")
+}
+
+/** Gets the textual representation of a summary component in the format used for MaD models. */
+string getMadRepresentationSpecific(SummaryComponent sc) {
+  exists(Content c |
+    sc = TContentSummaryComponent(c) and
+    result = getContentSpecific(c)
+  )
 }
 
-/** Gets the textual representation of a parameter position in the format used for flow summaries. */
```

Comment:
I am slightly confused by why the original text would be wrong. Is the point that the same representation is used for both? For flow summaries, I am thinking of `propagatesFlowExt`.

Reply:
At least for C# and Java the printing and parsing predicates are used for summaries, sources, sinks and neutrals - so saying that it was only for summaries would not be completely accurate.
Would you prefer that it is changed back to the original formulation for python?